src/plugins/outline/outline.tsx

```diff
@@ -0,0 +1,239 @@
+                      this.props.o().settings.showNotesInOutline
+                    ) {
+                      return (
+                        <li
+                          onClick={() => {
+                            this.jumpTo(model.id);
+                          }}
+                          class={() => getSelectedClass(model)}
+                        >
+                          <i class="dcg-icon-text"></i> {() => model.text ?? ""}
+                        </li>
+                      );
+                    } else if (model.type === "expression") {
+                      const exprModel = model;
+                      const colorLatexProperty = () =>
+                        exprModel?.formula?.rgb_value ??
+                        exprModel?.formula?.color_latex_value;
+                      const li = (
+                        <li
+                          onClick={() => {
+                            this.jumpTo(model.id);
+                          }}
+                          class={() => ({
+                            "dsm-outline-outline-default-expression": true,
+                            ...getSelectedClass(model),
+                            "dsm-outline-color-latex": !!colorLatexProperty(),
+                          })}
+                          style={() => {
+                            const shouldBeColored =
+                              (exprModel.formula?.is_graphable &&
+                                !exprModel.hidden) ??
+                              exprModel?.formula?.rgb_value;
+
+                            return {
+                              background: shouldBeColored
+                                ? colorLatexProperty()
+                                  ? Calc.getColorSwatchGradient?.(
+                                      (() => {
+                                        if (
+                                          Array.isArray(colorLatexProperty())
+                                        ) {
+                                          return {
+                                            type: "color-array",
+                                            value:
+                                              colorLatexProperty() as string[],
+                                          };
+                                        } else {
+                                          return {
+                                            type: "single-color",
+                                            value:
+                                              colorLatexProperty() as string,
+                                          };
+                                        }
+                                      })()
+                                    ).replace(
+                                      /[0-9.]+px/g,
+                                      (s) =>
+                                        `${
+                                          (Number(s.slice(0, -2)) *
```

Comment:
This is starting to get very indented. A few helper functions could save a ton here.